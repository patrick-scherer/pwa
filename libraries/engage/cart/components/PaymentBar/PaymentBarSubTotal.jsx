// @flow
import * as React from 'react';
import { SurroundPortals } from '@shopgate/engage/components';
import {
  CART_PAYMENT_BAR_TOTALS_SUB_TOTAL,
} from '@shopgate/pwa-common-commerce/cart/constants/Portals';
import CartTotalLine from '@shopgate/pwa-ui-shared/CartTotalLine';
import { CartContext } from '../../cart.context';
import { spacer } from './PaymentBarContent.style';
import connect from './PaymentBarSubTotal.connector';

type Props = {
  amount: number,
  label?: string,
  showSeparator?: boolean,
  className?: string,
}

/**
 * @returns {JSX}
 */
function PaymentBarSubTotal({
  amount, label, showSeparator, className,
}: Props) {
  const { currency, isLoading, hasPromotionCoupons } = React.useContext(CartContext);

  return (
    <SurroundPortals portalName={CART_PAYMENT_BAR_TOTALS_SUB_TOTAL}>
      {amount &&
        <CartTotalLine isDisabled={isLoading} type="subTotal" className={className}>
          <CartTotalLine.Label label={label} showSeparator={showSeparator} />
          <CartTotalLine.Amount amount={amount} currency={currency} />
          { hasPromotionCoupons && (
            <CartTotalLine.Spacer className={spacer} />
          )}
        </CartTotalLine>
      }
    </SurroundPortals>
  );
}

PaymentBarSubTotal.defaultProps = {
  className: null,
  showSeparator: true,
  label: 'cart.subtotal',
};

export default connect(PaymentBarSubTotal);
