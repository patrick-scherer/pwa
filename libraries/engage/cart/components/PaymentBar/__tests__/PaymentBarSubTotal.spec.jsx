import React from 'react';
import { shallow } from 'enzyme';
import PaymentBarSubTotal from '../PaymentBarSubTotal';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: () => ({
    currency: 'EUR',
  }),
}));
jest.mock('../PaymentBarSubTotal.connector', () => cmp => cmp);
jest.mock('@shopgate/engage/components');
jest.mock('@shopgate/pwa-core/helpers', () => ({
  logger: {
    error: jest.fn(),
  },
}));

describe('<PaymentBarSubTotal>', () => {
  it('should render line', () => {
    const wrapper = shallow(<PaymentBarSubTotal amount={10} />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
