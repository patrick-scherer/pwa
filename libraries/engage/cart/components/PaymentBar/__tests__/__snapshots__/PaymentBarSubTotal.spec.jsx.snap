// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentBarSubTotal> should render line 1`] = `
<CartTotalLine
  className={null}
  isDisabled={false}
  type="subTotal"
>
  <Label
    label="cart.subtotal"
    labelParams={Object {}}
    showSeparator={true}
    suffix={null}
  />
  <Amount
    amount={10}
    currency="EUR"
  />
</CartTotalLine>
`;
